### PR TITLE
Remove tenant_header_key from config

### LINKED ISSUE #2381: [tempo-query] Remove `tenant_header_key` config header option
It's highly misleading. While you can configure the plugin to use different headers to pass the tenant ID, Jaeger Query will only [propagate the bearer token](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/grpc/shared/grpc_client.go#L44) to the storage plugin. So, the only real option is using the default `bearer.token`. We should remove that config option.

### PR DESCRIPTION
**Pull request by kousikmitra**
**What this PR does**:

Removes `tenant_header_key` option from config

**Which issue(s) this PR fixes**:
Fixes #2381

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## main / unreleased
 
+* [ENHANCEMENT] Remove `tenant_header_key` option from `tempo-query` config [#2414](https://github.com/grafana/tempo/pull/2414) (@kousikmitra)
```
Comment (mdisibio) [inline CHANGELOG.md:3]: I think this should be `[CHANGE]`

**Review by mdisibio — CHANGES_REQUESTED** [review]
Hi thanks for the cleanup.  Just one small nit in the changelog and LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (kousikmitra) [reply CHANGELOG.md]: Updated

**Review by mapno — APPROVED** [review]
LGTM. Can you fix the conflict?

**Review by mdisibio — APPROVED** [review]

**Pull request merged**